shotover/src/hot_reload.rs

```diff
@@ -0,0 +1,77 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::os::unix::io::RawFd;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct FileDescriptor(pub RawFd);
+
+/// Requests that can be sent from replacement shotover to original shotover
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Request {
+    SendListeningSockets,
+    //BeginDrainingConnections,
+    //ShutdownOriginalNode,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum Response {
+    SendListeningSockets {
+        port_to_fd: HashMap<u32, FileDescriptor>,
+    },
+
+    //BeginDrainingConnections,
+    //ShutdownOriginalNode,
+    Error(String),
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::collections::HashMap;
+
+    #[test]
+
+    fn test_request_serialization() {
```

Comment:
I appreciate that you wrote tests here, but unfortunately what they are testing is "can serde_json round trip some values?", the `serde_json` library already has [many tests](https://github.com/serde-rs/json/tree/master/tests) answering this question, so we dont need to write our own.

I recommend removing these tests as keeping them would be a maintenance burden without any benefit.